lib/rom/sql/attribute.rb

```diff
@@ -36,7 +36,7 @@ def self.[](*args)
       # @api public
       def aliased(alias_name)
         super.with(name: name || alias_name).meta(
-          sql_expr: sql_expr.as(alias_name)
+          sql_expr: alias_sql_expr(sql_expr, alias_name)
```

Comment:
it looks like we should have our own composable wrapper for aliased expressions, that case analysis in `alias_sql_expr` is leaking

Reply:
I think this will be challenging to do in this PR, i'm not sure exactly how to approach it since from what i can see, `sql_expr` is an actual Sequel::* class everywhere so it may have unexpected breakage. Can we postpone this refactoring?